aten/src/ATen/native/cpu/IndexKernel.cpp

```diff
@@ -457,6 +457,64 @@ void masked_select_kernel(TensorIterator& iter, int64_t result_stride) {
     });
 }
 
+
+template <typename scalar_t>
+void cpu_hflip_vec(at::TensorIterator& iter) {
```

Comment:
Isn't there a way we can reuse `cpu_kernel_vec` for this op? Most probably the answer is "no", but it's not entirely clear the reason to me just yet.

Reply:
Sure, but my point is, can't we simply add an exception to the preconditions on that generic implementation so that `flip` would work with it?

Reply:
`vectorized_loop` doesn't accept any strides as arguments, so it would be a non-trivial change to the hottest path for CPU in pytorch.